app/actions/fileLoader.js

```diff
@@ -194,22 +205,31 @@ export function setStatsGamePage(index) {
 }
 
 export function deleteSelections(selections) {
-    return (dispatch, getState) => {
-      const tempStore = getState().fileLoader.fileLoadState;
-      const filesToRender = tempStore.filesToRender;
-      for(const i of selections){
-        shell.moveItemToTrash(i.fullPath);
-        filesToRender.splice(filesToRender.indexOf(i),1);
-      }
-      tempStore.filesToRender = filesToRender;
-      tempStore.filesOffset = filesToRender.length;
-      dispatch({
-        type: STORE_FILE_LOAD_STATE,
-        payload: {
-          fileLoadState: tempStore,
-        },
-      });
-    };
+  return (dispatch, getState) => {
+    const tempStore = getState().fileLoader.fileLoadState;
+    const filesToRender = tempStore.filesToRender;
+    _.each(selections, (selection, i) => {
+      shell.moveItemToTrash(selection.fullPath);
+      filesToRender.splice(i, 1);
```

Comment:
A few comments regarding this, we should probably almost always avoid using `splice()` since this modifies the array in place. Rather, we should probably either use `slice()` or even better, instead of constantly modifying the `filesToRender` file list with each iteration, we could probably just create a new list once, after moving all the items to trash.

Perhaps something like:
```
// do the iteration at the beginning
// then after that update the files list
filesToRender = tempStore.filesToRender.filter(() => /* something to filter out the selection */);
```